Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -21,49 +28,64 @@ module.exports.Quantity = class Quantity extends Expression
-      thisDurationInMilliseconds = durationInMilliseconds(@)
-      otherDurationInMilliseconds = durationInMilliseconds(other)
-      thisDurationInMilliseconds >= otherDurationInMilliseconds
-    else null
+    if other instanceof Quantity
+      other_v = ucum.convert(other.value,ucum_unit(other.unit),ucum_unit(@unit))
+      @value >= other_v
 
   after: (other) ->
-    if other instanceof Quantity and other.unit == @unit
-      @value > parseFloat other.value
-    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
-      thisDurationInMilliseconds = durationInMilliseconds(@)
-      otherDurationInMilliseconds = durationInMilliseconds(other)
-      thisDurationInMilliseconds > otherDurationInMilliseconds
-    else null
+    if other instanceof Quantity
+      other_v = ucum.convert(other.value,ucum_unit(other.unit),ucum_unit(@unit))
+      @value > other_v
 
   before: (other) ->
-    if other instanceof Quantity and other.unit == @unit
-      @value < parseFloat other.value
-    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
-      thisDurationInMilliseconds = durationInMilliseconds(@)
-      otherDurationInMilliseconds = durationInMilliseconds(other)
-      thisDurationInMilliseconds < otherDurationInMilliseconds
-    else null
+    if other instanceof Quantity
+      other_v = ucum.convert(other.value,ucum_unit(other.unit),ucum_unit(@unit))
+      @value < other_v
 
   equals: (other) ->
-    if other instanceof Quantity and @unit == other.unit
-      @value == parseFloat other.value
-    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
-      thisDurationInMilliseconds = durationInMilliseconds(@)
-      otherDurationInMilliseconds = durationInMilliseconds(other)
-      thisDurationInMilliseconds == otherDurationInMilliseconds
-    else null
+    if other instanceof Quantity
+      if (!@unit && other.unit ) || (@unit && !other.unit)
+        false
+      else if !@unit && !other.unit
+        @value == other.value
+      else
+        other_v = ucum.convert(other.value,ucum_unit(other.unit),ucum_unit(@unit))
+        decimalAdjust("round", @value, -8)  == decimalAdjust("round", other_v, -8)
+
+  convertUnits: (to_units) ->
+    convert_value(@value,@unit,to_units)
+
+  dividedBy: (other) ->
+    @multiplyDivied(other,"/")
+
+  multiplyBy: (other) ->
+    @multiplyDivied(other,".") # in ucum . represents multiplication
+
+  multiplyDivied: (other, operator) ->
+    if other instanceof Quantity
+      if @unit and other.unit
+        can_val = @to_ucum()
+        other_can_value = other.to_ucum()
+        ucum_value = ucum_multiply(can_val,[[operator,other_can_value]])
+        createQuantity(ucum_value.value, units_to_string(ucum_value.units))
+      else
+        value = if operator == "/" then @value / other.value  else @value * other.value
+        unit = @unit || other.unit
```

Comment:
Ah, yeah -- you're right.  The spec doesn't allow it.  Could you put in a comment noting that decimal / quantity should never happen, according to the spec?